### PR TITLE
code cleanup, target directory, global gitignore

### PR DESCRIPTION
**Pull request by sparr**

### CODE REVIEW

File: ZipMod.py [hunk 1]
```diff
@@ -37,32 +62,37 @@ def find_files(dir_path: str=None, patterns: [str]=None) -> [str]:
 
 # function to be passed to copytree, see https://docs.python.org/2/library/shutil.html#shutil.copytree
 def ignoreFunction(directory, files):
-  ignorePatterns = readGitIgnorePatterns()
+  ignorePatterns = readAllGitIgnorePatterns(modDir)
```
Comment (sparr) [inline ZipMod.py:65]: I am not happy about using a global variable here. Would be better to curry the ignoreFunction, but I'm too tired to code that right now.

**Comment by Pandemoneus** [comment]
The first three commits are fine to me, but I don't want to make the script any more complicated by including the user-global .gitignore.
By the way, I am not a Linux user, so the usr/bin/env is useless to me. I will keep it in for other Linux users though.

**Comment by sparr** [comment]
Without including the user-global .gitignore, you're going to end up with a lot of mods packaging .DS_store (mac) and Thumbs.db (windows) files. Since those are created by the OS and have nothing to do with the repo, they don't belong in the repo .gitignore file.

If you're on windows, the choice of shebang line does still have an effect, if you're using the official python launcher: https://docs.python.org/3/using/windows.html#shebang-lines

**Comment by Pandemoneus** [comment]
Well, I added all commits now because I suck at git. :P
Please also make sure you comment your code.

**Comment by sparr** [comment]
Will add comments in next PR!

**Pull request merged**